src/Service/ConnectionHandler.cpp

```diff
@@ -300,19 +315,21 @@ void ConnectionHandler::onSocketWritable(const Notification &)
                     destroyMe();
                     return;
                 }
-                continue;
+                copy_buffer_to_send();
             }
-
-            response->write(*send_buf);
-
-            packageSent();
-
-            if (send_buf->available() >= SENT_BUFFER_SIZE)
+            else
             {
-                send_buf->next();
+                WriteBufferFromFiFoBuffer buf;
+                response->write(buf);
+                out_buffer = buf.getBuffer();
+                copy_buffer_to_send();
             }
+            packageSent();
         }
-        /// If all sent remove writable event.
+
+        size_t sent = sock.sendBytes(send_buf);
```

Comment:
Consider to use `Socket::sendBytes(const SocketBufVec& buffers, int flags)` to avoid data copy.